### DISCUSSION TITLE
TorusWalletConnectorPlugin error on logout

### DISCUSSION THREAD

**Discussion by infocodiste** [Web3Auth/Web3Auth#693] [Help]
I am using TorusWalletConnectorPlugin.
On logout, the logout works but also gives me the following error:
"WalletLoginError: Wallet is not connectedNo wallet is connected"

**Comment by KetulCodiste** [comment]
Can anyone please answer this question quickly. It is an urgent requirement

**Reply by Felix-Asante** [reply]
I've also been having the same issue when I connect with social login or password less

**Comment by himanshuchawla009** [comment]
Hello can you attach your example code snippet/repo to look at, it happens if you calls logout function before connecting or when it is already logged out.

**Reply by infocodiste** [reply]
While using the TorusWalletConnectorPlugin, I got the following error
"base.esm.js:416 Uncaught (in promise) WalletLoginError: Wallet is not connectedNo wallet is connected
    at Function.value (base.esm.js:416:14)"
unfortunately I was not getting answer from the github discussion, I had to look through the core files to debug. 
But I have found the issue. This issue is in the plugin itself. 
When I click the logout button from my website, the internal logout function was executing twice and so I was getting the above error. It was executing second time because the TorusWalletConnectorPlugin was calling it again and thus giving the above error.


node_modules/@web3auth/torus-wallet-connector-plugin/dist/torusWalletConnectorPlugin.esm.js


key: "subscribeToWalletEvents",
    value: function subscribeToWalletEvents() {
      var _this$torusWalletInst,
          _this = this;

      (_this$torusWalletInst = this.torusWalletInstance) === null  _this$torusWalletInst === void 0 ? void 0 : _this$torusWalletInst.provider.on("accountsChanged", function () {
        var accounts = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];

        if (accounts.length === 0) {
          var _this$web3auth3;

          _this.torusWalletInstance.hideTorusButton();

          (_this$web3auth3 = _this.web3auth) === null  _this$web3auth3 === void 0 ? void 0 : _this$web3auth3.logout();
        }
      });
    }

This code was responsible for it. This error will be shown in the production website as well but in the console, so user experience will not hinder due to this bug but It would be better if this issue is solved so that the developers do not waste time to debug this error.

**Reply by infocodiste** [reply]
Can u please reply the question in ticket number #692. It is related to TorusWalletConnectorPlugin. I am trying hard to find a solution but was unable to find the solution till now.
